test/corecel/io/Logger.test.cc

```diff
@@ -20,8 +21,22 @@
 
 namespace celeritas
 {
+namespace detail
+{
 namespace test
 {
+TEST(NullLoggerMessage, all)
+{
+    NullLoggerMessage() << "This should not print anything " << range(10)
```

Comment:
Does this actually test that nothing is printed? (I.e. do we check the standard output against a references?)

Reply:
All the tests in this file are manually verified. This one in particular, though, shouldn't even compile if the null logger is working correctly: the `Range` object doesn't have an `operator<<`.

Reply:
> shouldn't even compile if the null logger is working correctly

That sounds bad/inverted ... it read that the build will fail to compile when things are right and compile and pass the test when things are wrong ... I must be missing something :)

Reply:
Right sorry, commenting at the middle of the night. It should fail to compile if it's not working :)

Reply:
Does the test runs on the device? (if not we would need to add such a test to check the main purpose of the `NullLoggerMessage`)

Reply:
I tested manually; it wouldn't be much of a test if we had one. The "no failure when streaming an unstreamable object" should be test enough...